homeassistant/components/sensor/yweather.py

```diff
@@ -0,0 +1,196 @@
+"""
+Support for the Yahoo! Weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.yweather/
+"""
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.const import (CONF_PLATFORM, TEMP_CELSIUS, TEMP_FAHRENHEIT,
+                                 CONF_MONITORED_CONDITIONS, STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ["yahooweather==0.3"]
+
+YAHOO_WEATHER = {}
+SENSOR_TYPES = {
+    'weather_current': ['Current', None],
+    'weather': ['Condition', None],
+    'temperature': ['Temperature', "temperature"],
+    'temp_min': ['Temperature', "temperature"],
+    'temp_max': ['Temperature', "temperature"],
+    'wind_speed': ['Wind speed', "speed"],
+    'humidity': ['Humidity', "%"],
+    'pressure': ['Pressure', "pressure"],
+    'visibility': ['Visibility', "distance"],
+}
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): "yweather",
+    vol.Required("woeid"): vol.Coerce(str),
+    vol.Optional("forecast", default=0): vol.Coerce(int),
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        [vol.In(SENSOR_TYPES.keys())],
+})
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yahoo! weather sensor."""
+    from yahooweather import UNIT_C, UNIT_F
+
+    unit = hass.config.temperature_unit
+    woeid = config.get("woeid", None)
+    forecast = config.get("forecast", 0)
+
+    if woeid is None:
+        _LOGGER.critical("You need woeid from yahoo!")
+        return False
+
+    if forecast > 4:
+        _LOGGER.error("Yahoo! don't support %d days forcast!", forecast)
+        return False
+
+    # convert unit
+    if unit == TEMP_CELSIUS:
+        yunit = UNIT_C
+    elif unit == TEMP_FAHRENHEIT:
+        yunit = UNIT_F
+    else:
+        _LOGGER.error("Unit %s is not supported by yahoo!", unit)
+        return False
+
+    # for print HA style temp
+    SENSOR_TYPES["temperature"][1] = unit
+    SENSOR_TYPES["temp_min"][1] = unit
+    SENSOR_TYPES["temp_max"][1] = unit
+
+    # exists a yahoo api object for this woeid?
+    yahoo_api = YAHOO_WEATHER.get(woeid, None)
+
+    # api for woeid don't exist
+    if yahoo_api is None:
+        yahoo_api = YahooWeatherData(woeid, yunit)
+        if not yahoo_api.update():
+            _LOGGER.critical("Can't retrieve data from yahoo!")
```

Comment:
Shouldn't we report here that there was no valid WOEID provided. I guess that `yahoo_api.update()` will give us a traceback at the moment with an invalid WOEID.